packages/react-components/react-datepicker-compat/src/components/DatePicker/DatePicker.test.tsx

```diff
@@ -136,4 +155,28 @@ describe('DatePicker', () => {
 
     expect(input.getAttribute('value')).toBe('15/1/20');
   });
+
+  it('calls onSelectDate when controlled', () => {
+    const onSelectDate = jest.fn();
+    const result = render(<ControlledDatePicker onSelectDate={date => onSelectDate()} />);
+
+    fireEvent.click(result.getByRole('combobox'));
+    result.getAllByRole('gridcell')[10].click();
+
+    expect(onSelectDate).toHaveBeenCalledTimes(1);
+  });
+
+  it('calls onSelectDate and formatDate when controlled', () => {
+    const onSelectDate = jest.fn();
+    const formatDate = jest.fn();
+    const result = render(
+      <ControlledDatePicker formatDate={date => formatDate()} onSelectDate={date => onSelectDate()} />,
```

Comment:
_nit:_ I think you can just pass `formatDate` here since `date` is optional in its definition.

```suggestion
      <ControlledDatePicker formatDate={formatDate} onSelectDate={date => onSelectDate()} />,
```